cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/prometheus/PrometheusMetricSamplerTest.java

```diff
@@ -184,18 +186,40 @@ public void testGetSamplesSuccess() throws Exception {
         addCapacityConfig(config);
         _prometheusMetricSampler.configure(config);
 
-        MetricSamplerOptions metricSamplerOptions = buildMetricSamplerOptions();
+        MetricSamplerOptions metricSamplerOptions = buildMetricSamplerOptions(TEST_TOPIC);
+        _prometheusMetricSampler._prometheusAdapter = _prometheusAdapter;
+
+        for (RawMetricType rawMetricType : _prometheusQueryMap.keySet()) {
+            setupPrometheusAdapterMock(rawMetricType, buildBrokerResults(),
+                buildTopicResults(TEST_TOPIC), buildPartitionResults(TEST_TOPIC));
+        }
+
+        replay(_prometheusAdapter);
+        MetricSampler.Samples samples = _prometheusMetricSampler.getSamples(metricSamplerOptions);
+
+        assertSamplesValid(samples, TEST_TOPIC);
+        verify(_prometheusAdapter);
+    }
+
+    @Test
+    public void testGetSamplesDotHandledTopic() throws Exception {
```

Comment:
This seems a little repetitive and I suspect that it might be useful to test dot handled topic with other cases.

* Would it be possible to keep `TEST_TOPIC` and `TEST_TOPIC_WITH_DOT` in a  `Set`, and then run each test in this class that requires a `topic` for each topic in our set?